modules/tests/kubecf-test.sh

```diff
@@ -96,10 +96,23 @@ kubectl logs -f "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --container "$co
 
 # Wait for the container to terminate and then exit the script with the container's exit code.
 jsonpath='{.status.containerStatuses[?(@.name == "'"$container_name"'")].state.terminated.exitCode}'
-while true; do
-  exit_code="$(kubectl get "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --output "jsonpath=${jsonpath}")"
-  if [[ -n "${exit_code}" ]]; then
-      exit "${exit_code}"
-  fi
-  sleep 1
+exit_code="$(kubectl get "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --output "jsonpath=${jsonpath}")"
+while [[ -z "${exit_code}" ]]; do
+    exit_code="$(kubectl get "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --output "jsonpath=${jsonpath}")"
+    sleep 1
 done
```

Comment:
(In addition to re-using the functions that I didn't realize existed) we could also just force things to loop at least once to make the code simpler:
```suggestion
exit_code=""
while [[ -z "${exit_code}" ]]; do
    exit_code="$(kubectl get "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --output "jsonpath=${jsonpath}")"
    sleep 1
done
```

Or we if we really want to not sleep excessively, we can do something like the old code:
```suggestion
while true; do
  exit_code="$(kubectl get "${pod_name}" --namespace "${KUBECF_NAMESPACE}" --output "jsonpath=${jsonpath}")"
  if [[ -n "${exit_code}" ]]; then
    break
  fi
  sleep 1
done
```

Reply:
Going with your first suggestion; we need to save the exit_code of the pod after the wait on the while anyways.